src/binder/textualidentityparser.cpp

```diff
@@ -417,12 +439,64 @@ namespace BINDER_SPACE
                 cursor++;
             }
 
-            if (dwFoundNumbers == static_cast<DWORD>(iRequiredVersionParts))
+            // Compat with .NET Framework:
+            // - If no value is provided, the version is considered invalid.
+            //   Example: "MyAssembly, Version="
+            // - If the major version is provided and the remaining components are valid, the version is considered valid.
+            //   Examples:
+            //     "MyAssembly, Version=1"
+            //     "MyAssembly, Version=1."
+            //     "MyAssembly, Version=1.."
+            //     "MyAssembly, Version=1..."
+            if (dwFoundNumbers == 0)
             {
-                pAssemblyVersion->SetFeatureVersion(dwNumbers[0], dwNumbers[1]);
-                pAssemblyVersion->SetServiceVersion(dwNumbers[2], dwNumbers[3]);
-                fIsValid = TRUE;
+                goto Exit;
             }
+
+            if (dwFoundNumbers < 2 || dwNumbers[0] == UnspecifiedNumber || dwNumbers[1] == UnspecifiedNumber)
+            {
+                // Compat with .NET Framework: If the major or minor version is unspecified, all components are considered
```

Comment:
Allowing less than 4 components for the version makes sense. 

But I do not think we should be matching the full .NET Framework bugs, like silently treating `Version=1` as unspecified version. The assembly loading works differently in CoreCLR, so we won't be able to match it 100% anyway.